File: SSUMobile/Modules/Radio/SSURadioModule.m
//
//  SSURadioModule.m
//  SSUMobile
//
//  Created by Eric Amorde on 9/8/15.
//  Copyright (c) 2015 Sonoma State University Department of Computer Science. All rights reserved.
//

#import "SSURadioModule.h"
#import "SSURadioConstants.h"
#import "SSUConfiguration.h"

@implementation SSURadioModule

+ (instancetype) sharedInstance {
    static SSURadioModule * instance = nil;
    static dispatch_once_t token;
    dispatch_once(&token, ^{
        instance = [[self alloc] init];
    });
    return instance;
}

#pragma mark - SSUModule

- (nonnull NSString *) title {
    return NSLocalizedString(@"Radio",
                             @"The campus online radio status - KSUN Radio");
}

- (nonnull NSString *) identifier {
    return @"radio";
}

- (UIView *) viewForHomeScreen {
    return [[UIImageView alloc] initWithImage:[UIImage imageNamed:@"radio_icon"]];
}

- (UIImage *) imageForHomeScreen {
    return [UIImage imageNamed:@"radio_icon"];
}

- (UIViewController *) initialViewController {
    UIStoryboard * storyboard = [UIStoryboard storyboardWithName:@"Radio_iPhone"
                                                          bundle:[NSBundle bundleForClass:[self class]]];
    return [storyboard instantiateInitialViewController];
}

- (BOOL) shouldNavigateToModule {
    if (![[SSUConfiguration sharedInstance] boolForKey:SSURadioStreamEnabledKey]) {
        /**
         KSUN radio has switched their streaming service to one which does not support
         mobile play. Unfortunately there is nothing we can do about this.
         So we will inform the user of this and go back
         */
        NSString * message = [[SSUConfiguration sharedInstance] stringForKey:SSURadioStreamDisabledMessageKey];
        UIAlertView * alert = [[UIAlertView alloc] initWithTitle:@"KSUN Currently Unavailable on Mobile"
                                                         message:message
                                                        delegate:nil
                                               cancelButtonTitle:@"Done"
                                               otherButtonTitles:nil];
        [alert show];
        return NO;
    }
    return YES;
}

- (BOOL) showModuleInNavigationBar {
    return NO;
}

@end

File: SSUMobile/Modules/Radio/SSURadioModule.swift
//
//  SSURadioModule.swift
//  SSUMobile
//
//  Created by Eric Amorde on 4/8/17.
//  Copyright © 2017 Sonoma State University Department of Computer Science. All rights reserved.
//

import Foundation
import UIKit

class SSURadioModule: SSUModuleBase, SSUModuleUI {
    
    @objc(sharedInstance)
    static let instance = SSURadioModule()
    
    // MARK: SSUModule
    
    var title: String {
        return NSLocalizedString("Radio", comment: "The campus online radio status - KSUN Radio")
    }
    
    var identifier: String {
        return "radio"
    }
    
    func setup() {
        
    }
    
    // MARK: SSUModuleUI
    
    func imageForHomeScreen() -> UIImage? {
        return UIImage(named: "radio_icon")
    }
    
    func initialViewController() -> UIViewController {
        let storyboard = UIStoryboard(name: "Radio_iPhone", bundle: Bundle(for: type(of: self)))
        return storyboard.instantiateInitialViewController()!
    }
    
    func shouldNavigateToModule() -> Bool {
        if !SSUConfiguration.sharedInstance().bool(forKey: SSURadioStreamEnabledKey) {
            let message = SSUConfiguration.sharedInstance().string(forKey: SSURadioStreamDisabledMessageKey)
            let alert = UIAlertController(title: "Radio Unavailable", message: message, preferredStyle: .alert)
            alert.addAction(UIAlertAction(title: "Done", style: .default, handler: nil))
            SSUGlobalNavigationController.sharedInstance().present(alert, animated: true, completion: nil)
            return false
        }
        return true
    }
}
